src/main/java/com/puppycrawl/tools/checkstyle/site/SiteUtil.java

```diff
@@ -832,32 +772,110 @@ else if (JAVADOC_TOKENS.equals(propertyName)) {
+                    "Missing @since on module " + moduleName);
+        }
+
+        DetailNode effectiveJavadoc = propertyJavadoc;
+        if (effectiveJavadoc == null) {
+            effectiveJavadoc = SUPER_CLASS_PROPERTIES_JAVADOCS.get(key);
+        }
+        final String propertySince;
+        if (effectiveJavadoc == null) {
+            propertySince = null;
+        }
+        else {
+            propertySince = getSinceVersionFromJavadoc(effectiveJavadoc);
+        }
+
+        final String result;
+        if (fallbackSince != null) {
+            result = fallbackSince;
         }
         else if (TOKENS.equals(propertyName)
-                        || JAVADOC_TOKENS.equals(propertyName)) {
-            // Use module's since version for inherited properties
-            sinceVersion = getSinceVersionFromJavadoc(moduleJavadoc);
+                || JAVADOC_TOKENS.equals(propertyName)) {
+            result = moduleSince;
+        }
+        else if (propertySince == null) {
+            result = moduleSince;
+        }
+        else if (isVersionAtLeast(propertySince, moduleSince)) {
+            result = propertySince;
         }
         else {
-            sinceVersion = getSinceVersionFromJavadoc(propertyJavadoc);
+            result = moduleSince;
         }
 
-        if (sinceVersion == null) {
-            final String message = String.format(Locale.ROOT,
-                    "Failed to find '@since' version for '%s' property"
-                            + " in '%s' and all parent classes.", propertyName, moduleName);
-            throw new MacroExecutionException(message);
+        return result;
+    }
+
+    /**
+     * Returns {@code true} if {@code actualVersion} ≥ {@code requiredVersion}.
+     * Both versions have any trailing "-SNAPSHOT" stripped before comparison.
+     *
+     * @param actualVersion   e.g. "8.3" or "8.3-SNAPSHOT"
+     * @param requiredVersion e.g. "8.3"
+     * @return {@code true} if, numerically, actualVersion is at least requiredVersion
+     */
+    private static boolean isVersionAtLeast(String actualVersion,
+                                            String requiredVersion) {
+        final String actClean = actualVersion
+                .replace(SNAPSHOT_SUFFIX, "")
+                .trim();
+        final String reqClean = requiredVersion
+                .replace(SNAPSHOT_SUFFIX, "")
+                .trim();
+
+        boolean result = false;
+        try {
+            final Version vAct = Version.parse(actClean);
+            final Version vReq = Version.parse(reqClean);
+            result = vAct.compareTo(vReq) >= 0;
+        }
+        catch (IllegalArgumentException iae) {
+            // Fallback: compare each numeric segment
+            final String[] partsAct = actClean.split(DOT_REGEX);
+            final String[] partsReq = reqClean.split(DOT_REGEX);
+            final int length = Math.max(partsAct.length, partsReq.length);
+
+            boolean compared = false;
```

Comment:
whole thing should be named well and hidden away as impl. detail.

separated concern could live in dedicated place (function).